app/ts/Workspace/WorkspaceContext.tsx

```diff
@@ -86,23 +111,153 @@ export function WorkspaceContextProviderComponent({
       .catch(errorCallback);
   }
 
+  function editTherapeuticContext(therapeuticContext: string): void {
+    const oldWorkspaceToSend: IOldWorkspace = _.merge(
+      {},
+      _.cloneDeep(oldWorkspace),
+      {
+        problem: {description: therapeuticContext}
+      }
+    );
+    setWorkspace(buildWorkspace(oldWorkspaceToSend, workspaceId, ordering));
+    sendOldWorkspace(oldWorkspaceToSend);
+  }
+
+  function editAlternative(alternative: IAlternative, newTitle: string): void {
+    const newAlternative: IAlternative = {id: alternative.id, title: newTitle};
+    const oldWorkspaceToSend: IOldWorkspace = _.merge(
+      {},
+      _.cloneDeep(oldWorkspace),
+      {problem: {alternatives: {[alternative.id]: newAlternative}}}
+    );
+    setWorkspace(buildWorkspace(oldWorkspaceToSend, workspaceId, ordering));
+    sendOldWorkspace(oldWorkspaceToSend);
+  }
+
+  function editCriterion(newCriterion: ICriterion): void {
+    const oldWorkspaceToSend: IOldWorkspace = _.merge(
```

Comment:
At some point I guess a good 'brain dead' task would be to find all _.merges and replace with schmear where appliccable